### DISCUSSION TITLE
Phalcon 4.2 PHP 7.4 Phalcon\Validation\Validator\File\Size\Min (wrong?) validation

### DISCUSSION THREAD

**Discussion by lefmic** [phalcon/cphalcon#16390] [Q&A]
Min file size validation seems to compare in the wrong direction:

```
$this->add(
    'thumbnail',
    new Min(
        [
            'size' => '1K',
            'message' => ':field is smaller than file size (:size)'
        ]
    )
);
```

Uploading a file of size 15K returns response: "thumbnail is smaller than file size (1K)".
Setting the value above 15K removes the error message.

I would expect this class to check that the file size is not less than 1K in this configuration. Did I miss something?

**Comment by niden** [comment]
@lefmic Can you please post a bit more of a verbose example? How do you check the upload?

**Reply by lefmic** [reply]
Sure!
I have a controller taking a request and passing the $_FILES array into the Validator. This works fine with the MaxSice validation and everything. Only the MinSize validation seems to validate in the same direction as the MaxSize.

MyController.php
```php
use Phalcon\Mvc\Controller;
use App\Validator\FileUploadValidation;

class MyController extends Controller
{
  public function uploadAction() {

    $validation = new FileUploadValidation();
    $messages = $validation->validate($_FILES);

    if ($messages->count() > 0) {
      $responseContent['result'] = 'error';
      foreach ($messages as $message) {
        $responseContent['errors'][] = $message->getMessage();
      }
      $response->setContent(json_encode($responseContent));
      return $response;
    }
  }
}
```

FileUploadValidation.php
```php
use Phalcon\Validation;
use Phalcon\Validation\Validator\File\Size\Min;

class FileUploadValidation extends Validation
{
  public function initialize()
  {
    $this->add(
      'thumbnail',
      new Min(
        [
          'size' => '1K',
          'message' => ':field is smaller than file size (:size)'
        ]
      )
    );
  }
}
```

With a file uploaded which is larger than 1K I expect no message to drop.
Now when I am testing this. I have no messages until I test with files which are larger than the size defined as minimum.

Here is an example:
Upload file 500bytes -> no messages
Upload file 50K -> message from validator appears

Message should be thrown only if file is smaller than the min size not bigger than the min size.

**Reply by niden** [reply]
@lefmic one more favor please. Can you just dump your $_FILES array before it gets validated?

I created one for a test to check this out and having trouble getting the validator to play nice. Sorry about that.

**Reply by lefmic** [reply]
Sorry for not responding for a while.
I have updated to phalcon 5 now but the error remains.
Here is the dump of $_FILES

`
array(1) {
  ["thumbnail"]=>
  array(6) {
    ["name"]=>
    string(23) "IMG-20180403-WA0000.jpg"
    ["full_path"]=>
    string(23) "IMG-20180403-WA0000.jpg"
    ["type"]=>
    string(10) "image/jpeg"
    ["tmp_name"]=>
    string(14) "/tmp/phpsjLIQJ"
    ["error"]=>
    int(0)
    ["size"]=>
    int(11768)
  }
}
`
Seems the Phalcon\Filter\Validation\Validator\File\Size\Max and Phalcon\Filter\Validation\Validator\File\Size\Min validators are doing the same thing.
Even the error messages in the code examples are the same: https://docs.phalcon.io/5.0/de-de/api/phalcon_filter#filter-validation-validator-file-size-min

**Comment by drowsyrearend** [comment]
> Sorry for not responding for a while. I have updated to phalcon 5 now but the error remains. Here is the dump of $_FILES
> 
> `array(1) { ["thumbnail"]=> array(6) { ["name"]=> string(23) "IMG-20180403-WA0000.jpg" ["full_path"]=> string(23) "IMG-20180403-WA0000.jpg" ["type"]=> string(10) "image/jpeg" ["tmp_name"]=> string(14) "/tmp/phpsjLIQJ" ["error"]=> int(0) ["size"]=> int(11768) } }` Seems the Phalcon\Filter\Validation\Validator\File\Size\Max and Phalcon\Filter\Validation\Validator\File\Size\Min validators are doing the same thing. Even the error messages in the code examples are the same: [https://docs.phalcon.io/5.0/de-de/api/phalcon_filter#filter-validation-validator-file-size-min](https://docs.phalcon.io/5.0/de-de/api/)https://moto-x3m.io/ [phalcon_filter#filter-validation-validator-file-size-min](https://docs.phalcon.io/5.0/de-de/api/phalcon_filter#filter-validation-validator-file-size-min)

love to know that, thx

**Comment by niden** [comment]
This has been resolved in https://github.com/phalcon/cphalcon/pull/16444

Thank you @lefmic
